zbus/src/object_server/interface.rs

```diff
@@ -130,6 +132,15 @@ pub trait Interface: Any + Send + Sync {
     fn introspect_to_writer(&self, writer: &mut dyn Write, level: usize);
 }
 
+/// A newtype for a reference counted Interface trait-object, with a custom Debug impl.
+#[derive(Clone)]
+pub(crate) struct ArcInterface(pub(crate) Arc<RwLock<dyn Interface>>);
+impl fmt::Debug for ArcInterface {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ArcInterface").finish_non_exhaustive()
```

Comment:
I don't think that faking the struct name in debug impl is a good idea. This is not `Interface` nor `Arc<RwLock<dyn Interface>>`, but an `ArcInterface` with _something_. We can rename `ArcInterface` to something like `InetrfaceObject`, or maybe you have better ideas.

> (especially since it's a private type).

There are a lot of other private types with debug impls that reveal their names. I don't see a problem with that.

Reply:
> I don't think that faking the struct name in debug impl is a good idea

It's not faking. The type is just an internal wrapper for `Arc<RwLock<dyn Interface>>` so it's not wrong to show that as the debug impl.

> > (especially since it's a private type).
> 
> There are a lot of other private types with debug impls that reveal their names. I don't see a problem with that.

True that it's not so important that we'll want to add manual `Debug` impl just to hide their names but I'm sure the derive impl, shows the internals as well (so anyone reading the debug repr would get the idea of what the type is about), not just the type name, like you're doing here.

Reply:
I won't object to `ArcInterface(Arc<RwLock<dyn Interface>>)` either but if we're manually implementing Debug anyway, might as well hide the name of the internal type that doesn't really matter.